src/app/shared/case-editor/case-edit-submit.component.ts

```diff
@@ -124,15 +142,28 @@ export class CaseEditSubmitComponent implements OnInit {
     for (let page of this.wizard.pages) {
       if (this.isShown(page)) {
         for (let field of page.case_fields) {
-          if (this.canShowField(field)) {
+          if (this.canShowFieldInCYA(field)) {
             // at least one field needs showing
             return true;
           }
         }
       }
     }
 
-    // found no fields to show in summary page
+    // found no fields to show in CYA summary page
+    return false;
+  }
+
+  readOnlySummaryfieldsToDisplayExists(): boolean {
+
+    for (let eventCaseField of this.eventTrigger.case_fields) {
```

Comment:
would this work?

return this.eventTrigger.case_fields.some(field => field.show_summary_change_option)

Reply:
perhaps we could also move this method on the CaseEventTrigger itself. Not sure though we are able to do that at the moment with the way we deserialise JSON objects

Reply:
Just did the first comment for now.